### PR TITLE
Use prefix caching default

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

For the missing `head_dim` I spent some time investigating `flashinfer` in order to support those head sizes, and it proved to be too much work for the value at the moment, it was much easier to sidestep prefix_caching/flashinfer for those.

Also most benefits (prefix caching/chunking) can also be implemented for `flashdecoding` so not all hope is lost.
There are 2 upsides to flashinfer atm:
- `block_size=1` means better/finer hitrate in the prefix caching.
- Support for fp8 kvcache.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -35,7 +35,7 @@ jobs:
         with:
           # Released on: 02 May, 2024
           # https://releases.rs/docs/1.78.0/
-          toolchain: 1.79.0
+          toolchain: 1.80.0
```
Comment (Narsil) [inline .github/workflows/tests.yaml:38]: Updated version of `bitstream-io` breaks on 1.79...

File: server/Makefile-flashinfer [hunk 2]
```diff
@@ -0,0 +1,2 @@
+install-flashinfer:
+	pip install flashinfer==0.1.5 -i https://flashinfer.ai/whl/cu124/torch2.4
```
Comment (drbh) [inline server/Makefile-flashinfer:2]: will this error on lower cuda versions? CI seems to be on 12.2 but also I'm not sure `make install-flashinfer` is included in the build commands. 

**tldr, just a bit confused about the flashinfer that will be used

File: launcher/src/main.rs [hunk 3]
```diff
@@ -1470,11 +1495,30 @@ fn main() -> Result<(), LauncherError> {
...
             let config: RawConfig = serde_json::from_str(&content)?;
 
-            if config.model_type == Some("gemma2".to_string()) {
-                tracing::info!("Forcing flash decoding because of softcap usage");
-                std::env::set_var("ATTENTION", "flashdecoding");
-            }
             let config: Config = config.into();
+            match config.head_dim {
+                Some(h) if h == 64 || h == 128 || h == 256 => {
+                    // std::env::set_var("ATTENTION", "flashdecoding");
```
Comment (drbh) [inline launcher/src/main.rs:1501]: nit
```suggestion
```

File: server/text_generation_server/layers/attention/cuda.py [hunk 4]
```diff
@@ -233,7 +233,7 @@ def attention(
         causal=True,
         softcap=0.0,
     ):
-        assert window_size_left == -1, "Windowing is not supported with flash infer"
+        # assert window_size_left == -1, "Windowing is not supported with flash infer"
```
Comment (drbh) [inline server/text_generation_server/layers/attention/cuda.py:236]: nit
```suggestion
```

**Review by drbh — DISMISSED** [review]
LGTM! added some nits and a question about library version above

File: server/Makefile-flashinfer (hunk 2 above)
Comment (Narsil) [reply server/Makefile-flashinfer:2]: CI ? the dockerfile also is upgraded to 12.4 actually (was done before to set the CI green again)

**Comment by Narsil** [comment]
Include in https://github.com/huggingface/text-generation-inference/pull/2449
